tests/test_config_flow.py

```diff
@@ -0,0 +1,163 @@
+    ).add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("pyhoma.client.TahomaClient.login", return_value=True), patch(
+        "custom_components.tahoma.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "custom_components.tahoma.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"username": "test@testdomain.com", "password": "test-password"},
+        )
+
+    assert result2["type"] == "abort"
+    assert result2["reason"] == "already_configured"
+
+async def test_allow_multiple_unique_entries(hass):
+    """Test config flow allows Config Flow unique entries."""
+    MockConfigEntry(
+        domain=config_flow.DOMAIN, unique_id="test2@testdomain.com", data={"username": "test2@testdomain.com", "password": "test-password"}
+    ).add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("pyhoma.client.TahomaClient.login", return_value=True), patch(
+        "custom_components.tahoma.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "custom_components.tahoma.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"username": "test@testdomain.com", "password": "test-password"},
+        )
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "test@testdomain.com"
+    assert result2["data"] == {"username": "test@testdomain.com", "password": "test-password"}
+
+async def test_import(hass):
+    """Test config flow using configuration.yaml."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    with patch("pyhoma.client.TahomaClient.login", return_value=True), patch(
+        "custom_components.tahoma.async_setup", return_value=True
+    ) as mock_setup, patch(
+        "custom_components.tahoma.async_setup_entry", return_value=True
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            config_flow.DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={"username": "test@testdomain.nl", "password": "test-password"},
+        )
+        assert result["type"] == "create_entry"
+        assert result["title"] == "test@testdomain.nl"
+        assert result["data"] == {
+            "username": "test@testdomain.nl",
+            "password": "test-password",
+        }
+
+        await hass.async_block_till_done()
+
+        assert len(mock_setup.mock_calls) == 1
+        assert len(mock_setup_entry.mock_calls) == 1
```

Comment:
Create a constant EMAIL_1 set to test@testdomain.com and EMAIL_2 to avoid to repeat them, it will ease the test also.